systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java

```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
 
-        List<Pod> pods = kubeClient(testStorage.getNamespaceName()).listPods(testStorage.getNamespaceName(), testStorage.getClusterName()).stream()
-                .filter(pod -> pod.getMetadata().getName().startsWith(testStorage.getClusterName()))
-                .collect(Collectors.toList());
+        LOGGER.info("Edit kafka labels in Kafka CR,as well as labels, and annotations of PVCs");
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), resource -> {
+            for (Map.Entry<String, String> label : customSpecifiedLabels.entrySet()) {
+                resource.getMetadata().getLabels().put(label.getKey(), label.getValue());
+            }
+            resource.getSpec().getKafka().getTemplate().getPersistentVolumeClaim().getMetadata().setLabels(customSpecifiedLabelOrAnnotationPvc);
+            resource.getSpec().getKafka().getTemplate().getPersistentVolumeClaim().getMetadata().setAnnotations(customSpecifiedLabelOrAnnotationPvc);
+            resource.getSpec().getZookeeper().getTemplate().getPersistentVolumeClaim().getMetadata().setLabels(customSpecifiedLabelOrAnnotationPvc);
+            resource.getSpec().getZookeeper().getTemplate().getPersistentVolumeClaim().getMetadata().setAnnotations(customSpecifiedLabelOrAnnotationPvc);
+        }, testStorage.getNamespaceName());
 
-        for (Pod pod : pods) {
-            LOGGER.info("Getting labels from {} pod", pod.getMetadata().getName());
-            verifyAppLabels(pod.getMetadata().getLabels());
-        }
+        LOGGER.info("Wait for rollout of zookeeper and kafka");
+        RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), testStorage.getZookeeperSelector(), 1, zkPods);
+        RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), 3, kafkaPods);
 
-        LOGGER.info("---> STATEFUL SETS <---");
+        LOGGER.info("---> PVC (both labels and annotation) <---");
 
-        Map<String, String> kafkaLabels = StrimziPodSetUtils.getLabelsOfStrimziPodSet(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName());
+        LOGGER.info("Wait for changes in PVC labels and until kafka becomes ready");
+        PersistentVolumeClaimUtils.waitUntilPVCLabelsChange(testStorage.getNamespaceName(), testStorage.getClusterName(), customSpecifiedLabelOrAnnotationPvc, pvcLabelOrAnnotationKey);
+        PersistentVolumeClaimUtils.waitUntilPVCAnnotationChange(testStorage.getNamespaceName(), testStorage.getClusterName(), customSpecifiedLabelOrAnnotationPvc, pvcLabelOrAnnotationKey);
 
-        LOGGER.info("Getting labels from stateful set of kafka resource");
-        verifyAppLabels(kafkaLabels);
+        pvcs = kubeClient(testStorage.getNamespaceName()).listPersistentVolumeClaims(testStorage.getNamespaceName(), testStorage.getClusterName()).stream().filter(
```

Comment:
```suggestion
        pvcs = kubeClient().listPersistentVolumeClaims(testStorage.getNamespaceName(), testStorage.getClusterName()).stream().filter(
```

+ plus same comment as above